src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/OAuthRequestUtils.java

```diff
@@ -36,7 +38,18 @@ private OAuthRequestUtils() {
      * @return true if the request is an OAuth request.
      */
     public static boolean isOAuthAccessAttempt(HttpServletRequest request) {
-        return is3LOAuthAccessAttempt(request) || is2LOAuthAccessAttempt(request);
+        return is3LOAuthAccessAttempt(request) || is2LOAuthAccessAttempt(request) || isOauthTokenRequest(request) || isOauthTokenRequest(request);
```

Comment:
this is also used by the `OAuth1aRequestFilter` maybe we should ask for a comment from @goravjoshi  to confirm that this won't have any unintended consequences for the general oauth flow?

Reply:
That's a really good point. I think it's better we just separate it into a different method and call both from the crumb exception. The other methods explicitly exclude request and access tokens

Reply:
Also the conditional is being tested twice... something weird happened with this commit...